src/Agent/NewRelic/Agent/Core/Transformers/TransactionTransformer/TransactionTransformer.cs

```diff
@@ -393,18 +393,20 @@ private void GenerateAndCollectSqlTrace(ImmutableTransaction immutableTransactio
 
         private void TryGenerateExplainPlans(IEnumerable<Segment> segments)
         {
-            if (!_configurationService.Configuration.SqlExplainPlansEnabled)
+            if (!_configurationService.Configuration.SqlExplainPlansEnabled
+                || (!_configurationService.Configuration.TransactionTracerEnabled && !_configurationService.Configuration.SlowSqlEnabled))
```

Comment:
This logic is a little convoluted to read. Suggestion: write a comment on the intention

Reply:
Added more comments.